src/runtime_src/xrt/device/hal2.cpp

```diff
@@ -231,27 +237,37 @@ void
 device::
 acquire_cu_context(const uuid& uuid,size_t cuidx,bool shared)
 {
-  if (m_handle && m_ops->mOpenContext) {
-    if (m_ops->mOpenContext(m_handle,uuid.get(),cuidx,shared))
-      throw std::runtime_error(std::string("failed to acquire CU(")
-                               + std::to_string(cuidx)
-                               + ") context '"
-                               + std::strerror(errno)
-                               + "'");
+  try {
+    get_core_device()->open_context(uuid, cuidx, shared);
+  }
+  catch (const xrt_core::system_error&) {
```

Comment:
I think you should catch `const std::exception&` to be just a bit broader before catching `...`, The std::exception may carry more information compared to default `...`.